### ISSUE TITLE
What does the European interoperability mean? (e.g. the case of Spain)

### ISSUE THREAD

**Issue by ouboub** [corona-warn-app/cwa-website#551]
Hi

I see that now Spain is added to the EU country, where the corona warn app should work. I'd like to clarify this subject. Suppose I am in Spain and I would like to continue using the corona warn app. So

1.  Interchanging keys:. In my understanding the corona warn app and the Spanish radar App would interchange keys, if the conditions are met:

    

-  being near to that person for longer than 15 min

- and being close than 1.5 meter.

2. Downloaded keys of positive cases. Would the corona warn app now when being in Spain, download the keys of positive test cases of the Spanish health system???

3. Positive test case. In case I have a positive test, will the app then connect and upload that information to the spanish health system?

thanks

**Comment by heinezen** [comment]
@ouboub @MikeMcC399 

I've moved the whole thread to cwa-website, so that we can discuss the FAQ changes.

I think @ouboub raises an important point that we have to clarify who the user should contact in case they want to be/have been tested. This is probably not obvious to everyone. However, I think this should get its own FAQ entry, so we do not pack one entry with too much information.

For the existing FAQ entry, we can include your suggestion @MikeMcC399 and tell the users where to find the info. You can go ahead and make a PR if you have the time.

We can also link the official EU Interoperability FAQ from the comission website: https://ec.europa.eu/commission/presscorner/detail/en/qanda_20_1905#gateway

----
Corona-Warn-App Open Source Team

**Comment by MikeMcC399** [comment]
@heinezen 
> For the existing FAQ entry, we can include your suggestion @MikeMcC399 and tell the users where to find the info. You can go ahead and make a PR if you have the time.

I will look doing this now. Thank you.

**Comment by ouboub** [comment]
> @ouboub @MikeMcC399
> 
> I've moved the whole thread to cwa-website, so that we can discuss the FAQ changes.
> 
> I think @ouboub raises an important point that we have to clarify who the user should contact in case they want to be/have been tested. This is probably not obvious to everyone. However, I think this should get its own FAQ entry, so we do not pack one entry with too much information.
> 

right, that sounds sensible, but please **add** a link to that particular FAQ. Sometimes relevant iinformation is 
somehow there, but difficult to find. 
The easier (to access) the better

**Comment by ouboub** [comment]
> We can also link the official EU Interoperability FAQ from the comission website: https://ec.europa.eu/commission/presscorner/detail/en/qanda_20_1905#gateway

I just read this document, as far as I can see, it **does not** address the problem whom to contact in case you have 
a positive test.

**Comment by MikeMcC399** [comment]
@heinezen 
I overestimated my ability to handle PRs in github unfortunately 😞 

If you can pick up my commits from 
https://github.com/corona-warn-app/cwa-website/pull/553/commits/99679e42aea350020ca04dc4810aaa430b369ccb and
https://github.com/corona-warn-app/cwa-website/pull/553/commits/af7d8bc72dbb9a82ad67daafbe3192f727018bf0

that would be great. Otherwise I can just post my suggested changes as text here.

When I get time, I will look out for github training 😀 so I am better prepared next time.

**Comment by MikeMcC399** [comment]
@ouboub 
> > We can also link the official EU Interoperability FAQ from the comission website: https://ec.europa.eu/commission/presscorner/detail/en/qanda_20_1905#gateway
> 
> I just read this document, as far as I can see, it **does not** address the problem whom to contact in case you have
> a positive test.

The EU FAQ document says in Section 3:

"What about if I did a test in another EU country?

You can only insert a positive coronavirus test result in the app of the country where the test was taken. However, when you enter the code in that app, thanks to the interoperability, citizens from the country that you have visited will get notified that they have been in close contact of an infected case."

Does that answer your point, or were you looking for something else?

**Comment by Ein-Tim** [comment]
@heinezen @MikeMcC399 I opened #554

**Comment by ouboub** [comment]
> @ouboub
> 
> > > We can also link the official EU Interoperability FAQ from the comission website: https://ec.europa.eu/commission/presscorner/detail/en/qanda_20_1905#gateway
> > 
> > 
> > I just read this document, as far as I can see, it **does not** address the problem whom to contact in case you have
> > a positive test.
> 
> The EU FAQ document says in Section 3:
> 
> "What about if I did a test in another EU country?
> 
> You can only insert a positive coronavirus test result in the app of the country where the test was taken. However, when you enter the code in that app, thanks to the interoperability, citizens from the country that you have visited will get notified that they have been in close contact of an infected case."
> 
> Does that answer your point, or were you looking for something else?

Thanks!
I see! So it was nearly at the end of the section, I did not see at my first read. 
I really recommend to copy this paragraph into the corona warn app FAQ, it is *very* important and should be easy to access.

sometimes the the documentation itself is not the problem it is the access to this information that needs to be improved.

**Comment by ouboub** [comment]
> @heinezen
> I overestimated my ability to handle PRs in github unfortunately 
> 
> If you can pick up my commits from
> [99679e4](https://github.com/corona-warn-app/cwa-website/commit/99679e42aea350020ca04dc4810aaa430b369ccb) and
> [af7d8bc](https://github.com/corona-warn-app/cwa-website/commit/af7d8bc72dbb9a82ad67daafbe3192f727018bf0)
> 
> that would be great. Otherwise I can just post my suggested changes as text here.
> 
> When I get time, I will look out for github training  so I am better prepared next time.

just a moment, does this mean you have write access?
how does this work? where to apply for that access.
Although I have basic knowledge of git, I have decent knowledge of mercurial and thanks to the 
excellent hg-git plugin, I can deal with git

**Comment by MikeMcC399** [comment]
@ouboub 
> just a moment, does this mean you have write access?
how does this work? where to apply for that access.

🤣 This is now the "blind leading the blind", but anyway ....
This is an open source project so the community doesn't get write access. If you want to contribute then you need to fork the repo, make your changes there and then create a pull request from your fork to the main repository. Then the right people from the Open Source Team can look at your request and approve, modify or reject it. This hasn't been discussed in the slack channel, but you might be interested in joining via https://covidapps.herokuapp.com/ to have a more interactive connection to active participants. As I said, I'm just going through github training myself so I understand how to use the tools properly.

**Comment by MikeMcC399** [comment]
@heinezen 

> I think @ouboub raises an important point that we have to clarify who the user should contact in case they want to be/have been tested. This is probably not obvious to everyone. However, I think this should get its own FAQ entry, so we do not pack one entry with too much information.

That is a good suggestion. I will open a new issue to track this point.

**Comment by ouboub** [comment]
> @ouboub
> 
> > just a moment, does this mean you have write access?
> > how does this work? where to apply for that access.
> 
>  This is now the "blind leading the blind", but anyway ....
> This is an open source project so the community doesn't get write access. If you want to contribute then you need to fork the repo, make your changes there and then create a pull request from your fork to the main repository. Then the right people from the Open Source Team can look at your request and approve, modify or reject it. This hasn't been discussed in the slack channel, but you might be interested in joining via https://covidapps.herokuapp.com/ to have a more interactive connection to active participants. As I said, I'm just going through github training myself so I understand how to use the tools properly.

@MikeMcC399 
thanks for the clarification, I see, thanks
(I know this mechanism, sigh, 
but frankly would have preferred to push directly to the repository, as I am used on other true open source projects GNU Emacs for example, or send patches via email.
In any case I understand that 
for this project the team found it more appropriate (the people from SAP, I suppose) to chose this slightly awkward approach.

**Comment by heinezen** [comment]
@ouboub @MikeMcC399 

I can assure you that the PR process on Github is the same for every other free software project and is not akward in any way. We cannot give the community write access to the repo because that would increase the chances of vandalization (not that expect this from you). The equivalent of Github write access would be getting the SSH keys for the GNU Emacs code repository server.

Besides the forking, the PR workflow is the same for the community and the development/open source team. There are only minor differences, e.g. that the PRs from the CWA team comes from a repo branch whereas community contributions come from a forked branch.

----
Corona Warn App Open Source Team

**Comment by ouboub** [comment]
> @ouboub @MikeMcC399
> 
> I can assure you that the PR process on Github is the same for every other free software project and is not akward in any way. We cannot give the community write access to the repo because that would increase the chances of vandalization (not that I expect this from you). The equivalent of Github write access would be getting the SSH keys for the GNU Emacs code repository server.
> 
> Besides the forking, the PR workflow is the same for the community and the development/open source team. There are only minor differences, e.g. that the PRs from the CWA team comes from a repo branch whereas community contributions come from a forked branch.
> 
> Corona Warn App Open Source Team

@heinezen 
That gets a bit off topic, but just to clarify.

1. GNU emacs is not on github. RMS would rather die (he is not the
  official maintainer anymore, but has enough influence).

2. In «classical» free and open software project, usually (since around 1984) people would submit patches and once the gained enough trust or reputation they would get write access.

I understand that this strategy might be complicated or impractical for the actual corona warn app, I just wanted to point out that github philosophy is is not equal free and open software. I called it awkward out of personal idiosyncrasy. Meanwhile Mike has published a contribution, so I better follow the advice of ESR, «shut up and show me your code»

**Comment by heinezen** [comment]
@ouboub 

Don't worry, I completely understand what you mean because I am from the free software community myself :) I just wanted to make sure that @MikeMcC399 (who is new to this) doesn't get the impression that he is doing something wrong or that we do not value contributions via forks. 

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
OT:

Best example for changes coming in via a Fork and now included in the 1.7.0 release from the iOS App:

https://github.com/corona-warn-app/cwa-app-ios/pull/1459 and https://github.com/corona-warn-app/cwa-app-ios/pull/1464

**Comment by ouboub** [comment]
> OT:
> 
> Best example for changes coming in via a Fork and now included in the 1.7.0 release from the iOS App:
> 
> [corona-warn-app/cwa-app-ios#1459](https://github.com/corona-warn-app/cwa-app-ios/pull/1459) and [corona-warn-app/cwa-app-ios#1464](https://github.com/corona-warn-app/cwa-app-ios/pull/1464)

time wrap? :)
I just downloaded the latest update today and it is 1.6, but I presume 1.7 is soon to be released

**Comment by Ein-Tim** [comment]
@ouboub 
No I don't think so, 1.7.0 RC1 (Release Candidate) has been released today, normally there are 7-9 Release Candidates for these major versions so I think this will be earliest released the beginning of December

**Comment by MikeMcC399** [comment]
The Robert-Koch-Institut has changed their advice about accepting the positive PCR test results from other countries. See https://github.com/corona-warn-app/cwa-website/issues/555#issuecomment-729661199.

**Comment by MikeMcC399** [comment]
The [CWA Solution Architecture Document](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md) has now been updated to include a section [Cross Border Interoperability](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#cross-border-interoperability) which explains how interoperability between different countries in Europe works.

**Comment by ouboub** [comment]
> The [CWA Solution Architecture Document](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md) has now been updated to include a section [Cross Border Interoperability](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#cross-border-interoperability) which explains how interoperability between different countries in Europe works.

thanks!
If I understand this correctly all the app that participate in this (inter european) project are able to interchange their keys via bluetooth, german<-->italian<-->spanish etc?
Correct

**Comment by MikeMcC399** [comment]
> If I understand this correctly all the app that participate in this (inter european) project are able to interchange their keys via bluetooth, german<-->italian<-->spanish etc?

Strictly speaking it is not the apps that exchange keys via Bluetooth, rather it is the GAEN (Google Apple Exposure Notification) Framework / System installed on the mobile devices which is working with Bluetooth. The apps (from whatever country, and providing they are compatible with GAEN) just switch the [Exposure Notification System (Google)](https://developers.google.com/android/exposure-notifications/exposure-notifications-api) or [Exposure Notification Framework (Apple)](https://developer.apple.com/documentation/exposurenotification) on or off, and communicate to this system layer via APIs. The apps do not send or receive Bluetooth signals themselves.

The bottom line is that if an App is using GAEN it is compatible with an app from another country using GAEN. As you know there is a list of these apps, together with their compatibility status on [the EU web site](https://ec.europa.eu/info/live-work-travel-eu/health/coronavirus-response/travel-during-coronavirus-pandemic/mobile-contact-tracing-apps-eu-member-states_en).

**Comment by ouboub** [comment]
Thanks!

**Comment by ouboub** [comment]
Ok, this is now off topic but what the heck:
(I opened an issue on https://github.com/corona-warn-app/cwa-app-ios)
Just today I receive a warning message, which started like «encounters with low risk» etc. (It did not inform me about the data of the encounter, I presume because reasons concerning the data protection). Be it as it may, where can I find documentation about the algorithm

1.  low
2.  medium
3.  high

The reason I am asking concerns other European corona app, in particula the Spanish one. That informs you about the date but does not provide a sophisticated warning system as the Corona Warn app. That is why I would like to bring the subject to the attention of the Covid Radar app, but I would need details about the algorithm.
Thanks and regards

**Comment by dsarkar** [comment]
Dear @ouboub, 

Re-opened, since this is an ongoing discussion.


Best wishes,
DS

---
Corona-Warn-App Open Source Team

**Comment by MikeMcC399** [comment]
If there are still open questions in this thread, could somebody please summarize them? I think we covered a lot of it already. Perhaps if there are still specific points open, it would be better to open a new issue to make the discussion clearer?

**Comment by heinezen** [comment]
Since we are getting further away from the original topic of this issue, I will close it now. If there are still follow-up questions, please create a new issue in https://github.com/corona-warn-app/cwa-documentation . This will also make the answer to these questions easier to find.

Thank you,
CH

----
Corona-Warn-App Open Source Team
